File: problem1.c
//Sum of all the multiples of 3 or 5 below a certain number.

#include <stdio.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    if(argc != 2)
    {
        printf("Proper usage: ./3or5 number");
    }

    int cutoff = atoi(argv[1]);
    int sum = 0;
    int threemultiples = (cutoff - 1) / 3;
    int fivemultiples = (cutoff - 1) / 5;
    do
    {
        sum = sum + (3 * threemultiples);
        threemultiples--;
    }
    while(threemultiples > 0);
    do
    {
        sum = sum + (5 * fivemultiples);
        fivemultiples--;
    }
    while(fivemultiples > 0);
    int sum15 = 0;
    int fifteenmultiples = (cutoff - 1) / 15;
    do
    {
        sum15 = sum15 + (15 * fifteenmultiples);
        fifteenmultiples--;
    }
    while (fifteenmultiples > 0);
    sum = sum - sum15;
    printf("Sum: %i\n", sum);
}

File: problem1.py
#Sum of all multiples of m1 or m2 below a certain n.

#return the sum of all multiples of m1 or m2 below a certain n.
#like 3or5.c, but able to account for any multiples and any n.


def all_multiples_below(m1, m2, n):
    """
    >>> all_multiples_below(3, 5, 22)
    104
    """
    return sum([i for i in range(n) if i % (m1 * m2) != 0 and (i % m1 == 0 or i % m2 == 0)])
